R/item_get_wfs.R

```diff
@@ -0,0 +1,83 @@
+#' @title Download and load from SB WFS service
+#' 
+#' @template manipulate_item
+#' 
+#' @import stringr
+#' 
+#' @description 
+#' This function attempts to download the spatial layer data attached to the
+#' requested SB item. SB exposes discrete spatial objects (points, polygons) as 
+#' WFS web services. This requires the GDAL libraries \code{rgdal} and \code{gdalUtils} 
+#' to be installed, which may require the external GDAL libraries as well.
+#' 
+#' @examples 
+#' \dontrun{
+#' library(sp)
+#' layer = item_get_wfs('55e372b9e4b05561fa208212')
+#' 
+#' plot(layer)
+#' 
+#' }
+#' 
+#' @export
+item_get_wfs = function(sb_id, ..., session){
+	
+	if(!requireNamespace("gdalUtils") || !requireNamespace("rgdal")){
+		stop('gdalUtils and rgdal packages not installed. 
+					Both are required to interact with WFS services. 
+				 	Please run: install.packages(c(\'rgdal\',\'gdalUtils\'))')
+	}
+	
+	wfs_url = item_get_wfs_url(sb_id)
+	
+	layer_info = gdalUtils::ogrinfo(paste0('WFS:', wfs_url))
+	
+	#from this layer info, we need to extract layer name
+	layer_raw = str_match_all(layer_info, '[[:digit:]]+: (sb:[[:alpha:]_]+) \\([Polygon|Multi Polygon|Points|Point]+\\)')
+	
+	layer_names = sapply(layer_raw, function(x)x[2])
+	
+	#drop the bounding box and footprint layers, which we don't want
+	layer_names = layer_names[!is.na(layer_names) & !layer_names %in% c('sb:boundingBox', 'sb:footprint')]
+	if(length(layer_names) > 1){
+		stop('SB Item WFS has > 1 layer. item_download_sp currently cannot handle more than one layer')
+	}
+	
+	fname = tempfile(fileext = '.shp')
+	
+	gdalUtils::ogr2ogr(wfs_url, fname, layer_names)
```

Comment:
@jread-usgs I'm pulling directly from the default URL for SB. It seems to be v1.0.0. Should I try forcing that to v1.1.0?